src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -28,6 +28,11 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
             }
 
             var pattern = BindPattern(node.Pattern, expressionType, hasErrors, diagnostics);
+            if (!hasErrors && pattern is BoundDeclarationPattern p && !p.IsVar && expression.ConstantValue == ConstantValue.Null)
```

Comment:
Why exclude `var` case?

Reply:
No, that would be a semantically breaking change. People currently do `if (expr is var tmp && ...` to introduce a temporary variable, and all that kind of code would break.

Reply:
I actually proposed making this change in dotnet/csharplang#792. The specific case which most concerns me is the following pattern:

```
switch (x)
{
case var _ when expression:
  break;
}
```